test_util/azure.py

```diff
@@ -0,0 +1,198 @@
+"""Working with azure effectively requires a few too many individual clients
+with rather specific APIs. This module is intended to give the most basic functionality
+to accomodate DC/OS testing
+"""
+import logging
+import re
+
+import azure.common.credentials
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.resource.resources import ResourceManagementClient
+from azure.mgmt.resource.resources.models import (DeploymentMode,
+                                                  DeploymentProperties,
+                                                  ResourceGroup, TemplateLink)
+from msrestazure.azure_exceptions import CloudError
+from retrying import retry
+
+from test_util.helpers import lazy_property
+
+
+# Very noisy loggers that do not help much in debug mode
+logging.getLogger("msrest").setLevel(logging.INFO)
+logging.getLogger("requests_oauthlib").setLevel(logging.INFO)
+log = logging.getLogger(__name__)
+
+# This interface is designed to only use a single deployment
+# Being as the azure interface is based around resource groups, deriving
+# deployment name from group names makes it easier to attach to creating deploys
+DEPLOYMENT_NAME = '{}-Deployment'
+
+
+class AzureWrapper:
+    def __init__(self, subscription_id, tenant_id, client_id, client_secret):
+        self.credentials = azure.common.credentials.ServicePrincipalCredentials(
+            client_id=client_id,
+            secret=client_secret,
+            tenant=tenant_id)
+        self.rmc = ResourceManagementClient(self.credentials, subscription_id)
+        self.nmc = NetworkManagementClient(self.credentials, subscription_id)
+
+
+class TemplateDeployment:
+    def __init__(self, group_name, azure_wrapper):
+        # each template deployment is a resource group
+        self.group_name = group_name
+        self.azure_wrapper = azure_wrapper
+
+    @classmethod
+    def create(
```

Comment:
Can we add type annotations here? Really hard to tell what many of these should be. is vm_size a str or an int? Same goes for things like agent_count, oauth_enabled.